x-pack/plugins/infra/public/hooks/use_get_saved_object.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useCallback } from 'react';
+import { SavedObjectAttributes, SimpleSavedObject } from 'src/core/public';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+export const useGetSavedObject = <SavedObjectType extends SavedObjectAttributes>(type: string) => {
+  const kibana = useKibana();
+  const [data, setData] = useState<SimpleSavedObject<SavedObjectType> | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const getObject = useCallback(
+    (id: string) => {
+      setLoading(true);
+      const fetchData = async () => {
+        try {
+          const savedObjectsClient = kibana.services.savedObjects?.client;
+          if (!savedObjectsClient) {
+            throw new Error('Saved objects client is unavailable');
+          }
+          const d = await savedObjectsClient.get<SavedObjectType>(type, id);
+          setError(null);
+          setLoading(false);
+          setData(d);
+        } catch (e) {
+          setLoading(false);
+          setError(e);
+        }
+      };
+      fetchData();
+    },
+    /* eslint-disable-next-line react-hooks/exhaustive-deps */
```

Comment:
What error is this suppressing?

Reply:
The linter just appears broken in this scenario, so not really sure.

Reply:
the linter is not smart enough to know that I need to depend on `kibana.services.savedObjects` because it appears in a nested function. The only way around it is to ignore it unfortunately.